src/python_testing/TC_ACL_2_4.py

```diff
@@ -0,0 +1,1731 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     factory-reset: true
+#     quiet: true
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --endpoint 1
+# === END CI TEST ARGUMENTS ===
+
+import random
+
+import chip.clusters as Clusters
+from chip.clusters.Types import Nullable, NullValue
+from chip.interaction_model import Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+# Add these constants at the top of the file after imports
+D_OK_EMPTY = bytes.fromhex('1718')
+D_OK_SINGLE = bytes.fromhex(
+    '17D00000F1FF01003D48656C6C6F20576F726C642E205468697320697320612073696E676C6520656C656D656E74206C6976696E6720617320612063686172737472696E670018')
+
+
+class TC_ACL_2_4(MatterBaseTest):
+    def desc_TC_ACL_2_4(self) -> str:
+        return "[TC-ACL-2.4] ACL "
+
+    def steps_TC_ACL_2_4(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "TH1 commissions DUT using admin node ID N1", is_commissioning=True),
+            TestStep(2, "TH1 reads DUT Endpoint 0 OperationalCredentials cluster CurrentFabricIndex attribute",
+                     "Result is SUCCESS, value is stored as F1"),
+            TestStep(3, "TH1 reads DUT Endpoint 0 AccessControl cluster ACL attribute",
+                     "Result is SUCCESS, value is list of AccessControlEntryStruct containing 1 element with admin entry N1"),
+            TestStep(4, "TH1 writes DUT Endpoint 0 AccessControl cluster ACL attribute with 3 elements (N1 admin, N2 view, N3 operate)", "Result is SUCCESS"),
+            TestStep(5, "TH1 reads DUT Endpoint 0 AccessControl cluster ACL attribute",
+                     "Result is SUCCESS, value is list of AccessControlEntryStruct containing 3 elements"),
+            TestStep(6, "TH1 writes DUT Endpoint 0 AccessControl cluster ACL attribute with modified entries (N1 admin, N2 manage, N3 admin)", "Result is SUCCESS"),
+            TestStep(7, "TH1 reads DUT Endpoint 0 AccessControl cluster ACL attribute",
+                     "Result is SUCCESS, value is list of AccessControlEntryStruct containing 3 elements with updated privileges"),
+            TestStep(8, "TH1 writes DUT Endpoint 0 AccessControl cluster ACL attribute with updated targets", "Result is SUCCESS"),
```

Comment:
```suggestion
            TestStep(8, "TH1 writes DUT Endpoint 0 AccessControl cluster ACL attribute with updated entries", "Result is SUCCESS"),
```
In my opinion, Since we are also updating Privilege and AuthMode